examples/vault-iam-auth/README.md

```diff
@@ -100,20 +113,91 @@ vault write \
+`boto3` AWS SDK library. This script is a modified version of the Python 2.x example
+posted by J. Thompson, the author of Vault's IAM auth method, at the Vault mailing
+list. It uses `boto3` to create a request to the AWS Security Token Service API
+with the action "GetCallerIdentity" and then signs the request using the AWS credentials.
+For more details on the IAM auth method, there's a talk by J. Thompson called
+[Deep Dive into Vault's AWS Auth Backend][talk].
+
+```bash
+signed_request=$(python /opt/vault/scripts/sign-request.py vault.service.consul)
+```
+
+Once we have the encrypted request created by the python script, we can pass it
+in the body of the login request we will send to the Vault Server.
+
+```
+iam_request_url=$(echo $signed_request | jq -r .iam_request_url)
+iam_request_body=$(echo $signed_request | jq -r .iam_request_body)
+iam_request_headers=$(echo $signed_request | jq -r .iam_request_headers)
+
+
+data=$(cat <<EOF
+{
+  "role":"$VAULT_ROLE_NAME",
+  "iam_http_request_method": "POST",
+  "iam_request_url": "$iam_request_url",
+  "iam_request_body": "$iam_request_body",
+  "iam_request_headers": "$iam_request_headers"
+}
+EOF
+)
+
+curl --request POST --data "$data" https://vault.service.consul:8200/v1/auth/aws/login"
+```
+
+After sending the login request to Vault, Vault will execute the STS request to
+verify the client's identity with AWS and return a JSON object with your login
+information containing the `client_token`. The client token is an ephemeral token
+that you will send with your future operations requests to Vault. It can expire,
+be rotated, or become invalid for some other reason and you might be required to
+authenticate again.
 
-To see the full script for authenticating check the [client user data script][user_data_auth_client].
+To see the full example script for authenticating, check the [client user data script][user_data_auth_client].
 
 
+#### With Vault cli tool
+
+If vault cli is installed we can perform the login operation with it. The `VAULT_ADDR`
+environment variable has to be set and you need to have AWS credentials in some form.
+The vault cli will look for credentials configured in the standard locations such as
+environment variables, ~/.aws/credentials, IAM instance profile, or ECS task role, in
+that order. The way the it works is the same as with an HTTP request. The vault cli
+tool uses the golang AWS SDK to the create the STS API request and sign it with the
+credentials for you.
+
+```bash
+# Example with getting credentials from instance metadata
+# The AWS session token is necessary here because these credentials are temporary
+creds=$(curl http://169.254.169.254/latest/meta-data/iam/security-credentials/<AWS-IAM-ROLE-NAME>)
+export AWS_ACCESS_KEY_ID=$(echo $creds | jq -r .AccessKeyId)
+export AWS_SECRET_ACCESS_KEY=$(echo $creds | jq -r .SecretAccessKey)
+export AWS_SESSION_TOKEN=$(echo $creds | jq -r .Token)
```

Comment:
Hm, are you sure you have to extract AWS creds from EC2 metadata here? All the AWS SDK tools I've ever used (which includes Vault) know to look for these creds automatically, with you having to set those env vars.